opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1106,6 +1109,9 @@ namespace Opm {
     BlackoilWellModel<TypeTag>::
     updateWellControlsAndNetwork(const bool mandatory_network_balance, const double dt, DeferredLogger& local_deferredLogger)
     {
+        // PJPE: calculate common THP for subsea manifold well group (item 3 of NODEPROP set to YES)
+        computeWellGroupThp(local_deferredLogger);
```

Comment:
I can be wrong in the long term. Ideally, this function should be called within the function `WellGroupHelpers::computeNetworkPressures()`. 

Can be the same location **near** where the following code lies, 

```c++
                    if (network.node(node).as_choke()){
                        // PJPE: Node pressure is set to the common THP of the wells
                        node_pressures[node] = group_state.well_group_thp(node);
```

Reply:
For now I moved it deeper and placed it in `updateWellControls` next to the call to `computeNetworkPressures()` which seems to improve performance.